src/Symfony/Component/HttpFoundation/RedirectResponse.php

```diff
@@ -41,6 +41,10 @@ public function __construct($url, $status = 302, $headers = array())
         if (!$this->isRedirect()) {
             throw new \InvalidArgumentException(sprintf('The HTTP status code is not a redirect ("%s" given).', $status));
         }
+
+        if (301 == $status && !array_key_exists('cache-control', $headers)) {
```

Comment:
I would instead move this into the main Response class just before sending the headers to the client.